Fix failing CI on stable v1.9.1 pytorch

## 🐛 Bug description

```
FAILED tests/ignite/distributed/test_launcher.py::test_check_idist_parallel_torch_launch_n_procs_gloo[tcp://0.0.0.0:22334]
```

https://github.com/pytorch/ignite/blob/053411d73047e278f22008f9f0f1775d0e51dcd5/tests/ignite/distributed/test_launcher.py#L96-L98

<details>
<summary>
Error message
</summary>

```text
----------------------------- Captured stdout call -----------------------------
b''b'/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated\nand will be removed in future. Use torch.distributed.run.\nNote that --use_env is set by default in torch.distributed.run.\nIf your script expects `--local_rank` argument to be set, please\nchange it to read from `os.environ[\'LOCAL_RANK\']` instead. See \nhttps://pytorch.org/docs/stable/distributed.html#launch-utility for \nfurther instructions\n\n  warnings.warn(\nWARNING:torch.distributed.run:*****************************************\nSetting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. \n*****************************************\nTraceback (most recent call last):\n  File "/home/runner/work/ignite/ignite/tests/ignite/distributed/check_idist_parallel.py", line 106, in <module>\n    with idist.Parallel(backend=args.backend, **dist_config) as parallel:\n  File "/home/runner/work/ignite/ignite/ignite/distributed/launcher.py", line 318, in __enter__\n    idist.initialize(self.backend, init_method=self.init_method)\n  File "/home/runner/work/ignite/ignite/ignite/distributed/utils.py", line 539, in initialize\n    _set_model(comp_model_cls(backend, **kwargs))\n  File "/home/runner/work/ignite/ignite/ignite/distributed/comp_models/native.py", line 94, in __init__\n    self._create_from_backend(\n  File "/home/runner/work/ignite/ignite/ignite/distributed/comp_models/native.py", line 126, in _create_from_backend\n    dist.init_process_group(backend, init_method=init_method, **init_pg_kwargs)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 534, in init_process_group\n    store, rank, world_size = next(rendezvous_iterator)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/rendezvous.py", line 184, in _tcp_rendezvous_handler\n    store = _create_c10d_store(result.hostname, result.port, rank, world_size, timeout)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/rendezvous.py", line 154, in _create_c10d_store\n    tcp_store = TCPStore(hostname, port, world_size, False, timeout)\nRuntimeError: connect() timed out.\nTraceback (most recent call last):\n  File "/home/runner/work/ignite/ignite/tests/ignite/distributed/check_idist_parallel.py", line 106, in <module>\nTraceback (most recent call last):\n  File "/home/runner/work/ignite/ignite/tests/ignite/distributed/check_idist_parallel.py", line 106, in <module>\n    with idist.Parallel(backend=args.backend, **dist_config) as parallel:\n  File "/home/runner/work/ignite/ignite/ignite/distributed/launcher.py", line 318, in __enter__\n    with idist.Parallel(backend=args.backend, **dist_config) as parallel:\n  File "/home/runner/work/ignite/ignite/ignite/distributed/launcher.py", line 318, in __enter__\n    idist.initialize(self.backend, init_method=self.init_method)\n      File "/home/runner/work/ignite/ignite/ignite/distributed/utils.py", line 539, in initialize\nidist.initialize(self.backend, init_method=self.init_method)\n  File "/home/runner/work/ignite/ignite/ignite/distributed/utils.py", line 539, in initialize\n    _set_model(comp_model_cls(backend, **kwargs))    \n_set_model(comp_model_cls(backend, **kwargs))\n  File "/home/runner/work/ignite/ignite/ignite/distributed/comp_models/native.py", line 94, in __init__\n  File "/home/runner/work/ignite/ignite/ignite/distributed/comp_models/native.py", line 94, in __init__\n    self._create_from_backend(    \nself._create_from_backend(\n  File "/home/runner/work/ignite/ignite/ignite/distributed/comp_models/native.py", line 126, in _create_from_backend\n  File "/home/runner/work/ignite/ignite/ignite/distributed/comp_models/native.py", line 126, in _create_from_backend\n    dist.init_process_group(backend, init_method=init_method, **init_pg_kwargs)    \ndist.init_process_group(backend, init_method=init_method, **init_pg_kwargs)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 534, in init_process_group\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 534, in init_process_group\n        store, rank, world_size = next(rendezvous_iterator)store, rank, world_size = next(rendezvous_iterator)\n\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/rendezvous.py", line 184, in _tcp_rendezvous_handler\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/rendezvous.py", line 184, in _tcp_rendezvous_handler\n    store = _create_c10d_store(result.hostname, result.port, rank, world_size, timeout)    store = _create_c10d_store(result.hostname, result.port, rank, world_size, timeout)\n\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/rendezvous.py", line 154, in _create_c10d_store\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/rendezvous.py", line 154, in _create_c10d_store\n        tcp_store = TCPStore(hostname, port, world_size, False, timeout)tcp_store = TCPStore(hostname, port, world_size, False, timeout)\n\nRuntimeErrorRuntimeError: : connect() timed out.connect() timed out.\n\nTraceback (most recent call last):\n  File "/home/runner/work/ignite/ignite/tests/ignite/distributed/check_idist_parallel.py", line 106, in <module>\n    with idist.Parallel(backend=args.backend, **dist_config) as parallel:\n  File "/home/runner/work/ignite/ignite/ignite/distributed/launcher.py", line 318, in __enter__\n    idist.initialize(self.backend, init_method=self.init_method)\n  File "/home/runner/work/ignite/ignite/ignite/distributed/utils.py", line 539, in initialize\n    _set_model(comp_model_cls(backend, **kwargs))\n  File "/home/runner/work/ignite/ignite/ignite/distributed/comp_models/native.py", line 94, in __init__\n    self._create_from_backend(\n  File "/home/runner/work/ignite/ignite/ignite/distributed/comp_models/native.py", line 126, in _create_from_backend\n    dist.init_process_group(backend, init_method=init_method, **init_pg_kwargs)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 534, in init_process_group\n    store, rank, world_size = next(rendezvous_iterator)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/rendezvous.py", line 184, in _tcp_rendezvous_handler\n    store = _create_c10d_store(result.hostname, result.port, rank, world_size, timeout)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/rendezvous.py", line 154, in _create_c10d_store\n    tcp_store = TCPStore(hostname, port, world_size, False, timeout)\nRuntimeError: connect() timed out.\nERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 11321) of binary: /usr/share/miniconda3/envs/test/bin/python\nTraceback (most recent call last):\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/runpy.py", line 194, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/runpy.py", line 87, in _run_code\n    exec(code, run_globals)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/launch.py", line 193, in <module>\n    main()\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/launch.py", line 189, in main\n    launch(args)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/launch.py", line 174, in launch\n    run(args)\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/run.py", line 689, in run\n    elastic_launch(\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 116, in __call__\n    return launch_agent(self._config, self._entrypoint, list(args))\n  File "/usr/share/miniconda3/envs/test/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 244, in launch_agent\n    raise ChildFailedError(\ntorch.distributed.elastic.multiprocessing.errors.ChildFailedError: \n*******************************************************************************************\n  /home/runner/work/ignite/ignite/tests/ignite/distributed/check_idist_parallel.py FAILED  \n===========================================================================================\nRoot Cause:\n[0]:\n  time: 2021-09-22_16:18:17\n  rank: 0 (local_rank: 0)\n  exitcode: 1 (pid: 11321)\n  error_file: <N/A>\n  msg: "Process failed with exitcode 1"\n===========================================================================================\nOther Failures:\n[1]:\n  time: 2021-09-22_16:18:17\n  rank: 1 (local_rank: 1)\n  exitcode: 1 (pid: 11322)\n  error_file: <N/A>\n  msg: "Process failed with exitcode 1"\n[2]:\n  time: 2021-09-22_16:18:17\n  rank: 2 (local_rank: 2)\n  exitcode: 1 (pid: 11323)\n  error_file: <N/A>\n  msg: "Process failed with exitcode 1"\n[3]:\n  time: 2021-09-22_16:18:17\n  rank: 3 (local_rank: 3)\n  exitcode: 1 (pid: 11324)\n  error_file: <N/A>\n  msg: "Process failed with exitcode 1"\n*******************************************************************************************\n\n'
```

</details>

--- a/tests/ignite/distributed/test_auto.py
+++ b/tests/ignite/distributed/test_auto.py
@@ -204,8 +204,7 @@ def test_auto_methods_gloo(distributed_context_single_node_gloo):
     _test_auto_model_optimizer(ws, device)
 
     if ws > 1 and device.type == "cpu":
-        error_type = AssertionError if LooseVersion(torch.__version__) <= LooseVersion("1.9.0") else ValueError
-        with pytest.raises(error_type, match=r"SyncBatchNorm layers only work with GPU modules"):
+        with pytest.raises(AssertionError, match=r"SyncBatchNorm layers only work with GPU modules"):
             model = nn.Sequential(nn.Linear(20, 100), nn.BatchNorm1d(100))
             auto_model(model, sync_bn=True)
 

--- a/tests/ignite/distributed/test_auto.py
+++ b/tests/ignite/distributed/test_auto.py
@@ -1,5 +1,4 @@
 import os
-from distutils.version import LooseVersion
 
 import pytest
 import torch

--- a/tests/ignite/distributed/test_auto.py
+++ b/tests/ignite/distributed/test_auto.py
@@ -1,4 +1,5 @@
 import os
+from distutils.version import LooseVersion
 
 import pytest
 import torch
@@ -203,7 +204,8 @@ def test_auto_methods_gloo(distributed_context_single_node_gloo):
     _test_auto_model_optimizer(ws, device)
 
     if ws > 1 and device.type == "cpu":
-        with pytest.raises(AssertionError, match=r"SyncBatchNorm layers only work with GPU modules"):
+        error_type = AssertionError if "dev" not in torch.__version__ else ValueError
+        with pytest.raises(error_type, match=r"SyncBatchNorm layers only work with GPU modules"):
             model = nn.Sequential(nn.Linear(20, 100), nn.BatchNorm1d(100))
             auto_model(model, sync_bn=True)
 

--- a/tests/ignite/distributed/test_launcher.py
+++ b/tests/ignite/distributed/test_launcher.py
@@ -87,7 +87,7 @@ def _test_check_idist_parallel_torch_launch(init_method, fp, backend, nprocs):
 @pytest.mark.distributed
 @pytest.mark.skipif(not has_native_dist_support, reason="Skip if no native dist support")
 @pytest.mark.skipif("WORLD_SIZE" in os.environ, reason="Skip because test uses torch launch")
-@pytest.mark.parametrize("init_method", [None, "tcp://0.0.0.0:22334", "FILE"])
+@pytest.mark.parametrize("init_method", [None, "tcp://0.0.0.0:29500", "FILE"])
 def test_check_idist_parallel_torch_launch_n_procs_gloo(init_method, dirname, exec_filepath):
     if init_method == "FILE":
         init_method = f"file://{dirname}/shared"
@@ -102,7 +102,7 @@ def test_check_idist_parallel_torch_launch_n_procs_gloo(init_method, dirname, ex
 @pytest.mark.skipif(not has_native_dist_support, reason="Skip if no native dist support")
 @pytest.mark.skipif("WORLD_SIZE" in os.environ, reason="Skip because test uses torch launch")
 @pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU")
-@pytest.mark.parametrize("init_method", [None, "tcp://0.0.0.0:22334", "FILE"])
+@pytest.mark.parametrize("init_method", [None, "tcp://0.0.0.0:29500", "FILE"])
 def test_check_idist_parallel_torch_launch_n_procs_nccl(init_method, dirname, exec_filepath):
     if init_method == "FILE":
         init_method = f"file://{dirname}/shared"
@@ -198,7 +198,7 @@ def _test_func(index, ws, device, backend, true_init_method):
 @pytest.mark.distributed
 @pytest.mark.skipif("WORLD_SIZE" in os.environ, reason="Skip if launched as multiproc")
 @pytest.mark.skipif(not has_native_dist_support, reason="Skip if no native dist support")
-@pytest.mark.parametrize("init_method", ["env://", "tcp://0.0.0.0:22334", "FILE"])
+@pytest.mark.parametrize("init_method", ["env://", "tcp://0.0.0.0:29500", "FILE"])
 @pytest.mark.parametrize(
     "backend",
     ["gloo", pytest.param("nccl", marks=pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU"))],
@@ -216,7 +216,7 @@ def test_idist_parallel_spawn_n_procs_native(init_method, backend, dirname):
 @pytest.mark.distributed
 @pytest.mark.skipif("WORLD_SIZE" not in os.environ, reason="Skip if not launched as multiproc")
 @pytest.mark.skipif(not has_native_dist_support, reason="Skip if no native dist support")
-@pytest.mark.parametrize("init_method", ["env://", "tcp://0.0.0.0:22334", "FILE"])
+@pytest.mark.parametrize("init_method", ["env://", "tcp://0.0.0.0:29500", "FILE"])
 @pytest.mark.parametrize(
     "backend",
     ["gloo", pytest.param("nccl", marks=pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU"))],

